jax/_src/lax/lax.py

```diff
@@ -113,16 +115,16 @@ def _try_broadcast_shapes(shapes):
   return tuple(result_shape)
 
 @cache()
-def broadcast_shapes(*shapes):
+def broadcast_shapes(*shapes: Tuple[int, ...]) -> Tuple[int, ...]:
   """Returns the shape that results from NumPy broadcasting of `shapes`."""
   if len(shapes) == 1:
     return shapes[0]
   ndim = _max(len(shape) for shape in shapes)
-  shapes = [(1,) * (ndim - len(shape)) + shape for shape in shapes]
-  result_shape = _try_broadcast_shapes(shapes)
+  shape_list = [(1,) * (ndim - len(shape)) + shape for shape in shapes]
+  result_shape = _try_broadcast_shapes(shape_list)
   if result_shape is None:
     raise ValueError("Incompatible shapes for broadcasting: {}"
-                     .format(tuple(map(tuple, shapes))))
+                     .format(tuple(shape_list)))
```

Comment:
I modified the error display here because MyPy was [having trouble](https://mypy-play.net/?mypy=latest&python=3.10&gist=ec51e527a58bd719a3bc049e443c9c90).  However, I don't think `map` is necessary since if `shape_list` contains anything other than tuples, the tuple addition on line 123 won't work.